Add more preview icons for github issues and PRs

choldgraf opened the pull request:
This adds a few more icons for the hover previews with github issues and PRs. Because we have a bunch of icons now, it moves some of the logic for choosing the right icon above the part where we actually render it. The other changes are just minor naming updates so that it's easier to understand what the variables are doing. I added a little docs page so we can preview the outputs.

I had to make some arbitrary decisions about the icons to use, since we're using heroicons and some of them don't quite have the same icons that github uses, so I'm happy to switch to other icon choices if those are better.

Demo page:

https://deploy-preview-747--myst-theme.netlify.app/reference/github-previews/

Here's a video of what they look like:

https://github.com/user-attachments/assets/7f606060-8d54-40ad-b8b0-bd06c49f8d98

- closes jupyter-book/myst-theme#746.

agoose77 commented:
This looks really useful! I think we should create a test page for this in the docs, and create some test issues / PRs somewhere (probably a new repo, for ease) that we can use to visually spot check it. I can get to that next week / at the weekend.

choldgraf commented:
OK I've done these things:

- Added a test repository here: https://github.com/jupyter-book/test-repo
- Created several issues and PRs of the various states we want
- Updated our table to use those instead of the grab-bag we had before

Check out the table preview in the docs
